pkg/smartcontract/manifest/group.go

```diff
@@ -38,11 +38,14 @@ func (g *Group) IsValid(h util.Uint160) error {
 }
 
 // AreValid checks for groups correctness and uniqueness.
+// If the contract hash is empty, then hash-related checks are omitted.
 func (g Groups) AreValid(h util.Uint160) error {
-	for i := range g {
-		err := g[i].IsValid(h)
-		if err != nil {
-			return err
+	if h.Equals(util.Uint160{}) {
```

Comment:
Also, may be it is better to have this logic in the compiler itself? Otherwise we can miss real errors.

Reply:
> Shouldn't the condition be inverted?

Nice catch, fixed and the test is added.

> Otherwise we can miss real errors.

In real blockchain life `(m *Manifest) IsValid(hash util.Uint160) error ` never accepts empty hash, so extending it with empty hash case will not affect real cases.